recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,136 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class Mosquitto(ConanFile):
+    name = "mosquitto"
+    license = "EPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mosquitto.org"
+    description = """Eclipse Mosquitto MQTT library, broker and more"""
+    topics = ("MQTT", "IoT", "eclipse")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+                "with_tls": [True, False],
+                "clients": [True, False],
+                "broker": [True, False],
+                "apps": [True, False],
+                "plugins": [True, False],
+#                 "with_cjson": [True, False], TODO , adding a dependency is not enough, it needs also be found
+            }
+    default_options = {"shared": False,
+                        "with_tls": True,
+                        "clients": False,
+                        "broker": False,
+                        "apps": False,
+                        "plugins": False,  # TODO, there is some logic, just enabling plugin does not work, needs also something else
+#                        "with_cjson": False , TODO , adding a dependency is not enough, it needs also be found
+    }
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_tls:
+            self.requires("openssl/1.1.1i")
+        # TODO
+        # if self.options.with_cjson:
+        #     self.requires("cjson/1.7.14")
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("Visual Studio build for any MT runtime is not supported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_THREADING"] = self.settings.os != "Windows"
+        self._cmake.definitions["WITH_PIC"] = self.settings.os != "Windows"
```

Comment:
This is the bit that requires nasty patches or was that a different project?

Reply:
not this project